src/server/editorServices.ts

```diff
@@ -464,6 +464,11 @@ namespace ts.server {
         return copiedList;
     }
 
+    export function forEachProject<T>(projects: Project[], action: (project: Project) => T[], comparer?: (a: T, b: T) => number, areEqual?: (a: T, b: T) => boolean) {

```

Comment:
Indeed that's a better name. The other `forEach`, I often mistaken it as the an `iterate`, only find out later that not all elements are traversed because something has a falsy value.